Update `@woocommerce/extend-checkout-block` to include an example of a forced inner block

opr opened the pull request:
Blocked until https://github.com/woocommerce/woocommerce-blocks/pull/6763 is merged

### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
This PR will update the `extend-checkout-block` package. The updates included here are:

- Add an example of a forced inner block to the WooCommerce Blocks Checkout Block (It will appear as an inner block of the Contact Information Block)
- Refactor the main integration file to split the script/style registration into inner block/main integration.
- Fix the `Internal Dependencies` section of `src/js/index.js.mustache`
<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

### How to test the changes in this Pull Request:

1. Create a (local) test site. Set up WooCommerce and import products.
2. Clone [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) and ensure you're on `trunk`.
3. Build WooCommerce Blocks (instructions to do so are in the [repo readme](https://github.com/woocommerce/woocommerce-blocks)) and activate the plugin.
4. Create a page and add the Checkout Block. Notice the Contact Information Block within it doesn't have anything under the email box.
5. Save the page and keep this tab open for later.
6. Go to the `wp-content/plugins` directory and run `npx @wordpress/create-block -t ./woocommerce-monorepo/packages/js/extend-cart-checkout-block/ some-extension-name` (note you will need to change the path to this package if your WC repo is installed elsewhere, but this must be run from `wp-content/plugins`)
7. Go to the WP Dashboard -> Plugins and activate the `Some Extension Name` plugin.
8. Reload the editor page with the Checkout Block from step 4 and 5.
9. Observe the newsletter signup checkbox is now present. Ensure there is text accompanying it. ⚠️ **Do not re-save the page yet!**
10. Add some items to your cart then load the Checkout Block page on the front-end.
11. Ensure the newsletter signup checkbox is showing. Toggle it and ensure the validation warning shows.
12. Go back to the Checkout page in the editor and save it. Reload the page after successfully saving and ensure the checkbox is there.
13. Load it on the frontend and ensure the checkbox is there.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable? **Not applicable, manual testing is OK**
-   [x] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`? **I added one by using `./vendor/bin/changelogger add`, but running `./vendor/bin/changelogger write --use-version=1.1.0` did nothing. When I ran the mentioned `pnpm` script it says `None of the selected packages has a "changelog" script` when I run it. I also don't know if I need to add an entry to `CHANGELOG.md` - please let me know if I do!**

📝 Note: this package is not going to be installed by `npm install` it is used only with `@wordpress/create-block` which is why I marked it as a patch version in the changelog.

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

opr committed:
Update main integration file to register newsletter block scripts/styles

opr committed:
Add files for example newsletter block

opr committed:
Use only one internal dependencies comment

opr committed:
Include newsletter block in index

opr committed:
Move entry to correct position in webpack

opr committed:
Rename functions to register scripts and styles

opr committed:
Add changelog

opr commented:
cc @PanosSynetos you could subscribe to this issue so you know when it's merged if you like!

psealock reviewed (commented):
This looks really cool @opr. This serves a similar purpose as our [create extension script](https://github.com/woocommerce/woocommerce/blob/trunk/tools/create-extension/README.md), but a much better delivery mechanism.

> Notice the Contact Information Block within it doesn't have anything under the email box.

I'm getting an email placed in the text box

![image](https://user-images.githubusercontent.com/1922453/182995930-f6dfce98-24fc-459d-bc63-657374479c14.png)

Upon activating `some-extension-name`, I get the following error which is preventing me from continuing. Does it look familiar? I'm not sure why this would happen.

```
Notice: Function WP_Block_Type_Registry::register was called incorrectly. Block type names must contain a namespace prefix. Example: my-plugin/my-custom-block-type Please see [Debugging in WordPress](https://wordpress.org/support/article/debugging-in-wordpress/) for more information. (This message was added in version 5.0.0.) in /var/www/html/wp-includes/functions.php on line 5831 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-includes/functions.php:5831) in /var/www/html/wp-admin/includes/misc.php on line 1381 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-includes/functions.php:5831) in /var/www/html/wp-includes/functions.php on line 6878 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-includes/functions.php:5831) in /var/www/html/wp-admin/admin-header.php on line 9 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-includes/functions.php:5831) in /var/www/html/wp-includes/option.php on line 1112 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-includes/functions.php:5831) in /var/www/html/wp-includes/option.php on line 1113
```
Here are my plugins, for reference

```
+---------------------+----------+------------------------------+-----------+
| name                | status   | update                       | version   |
+---------------------+----------+------------------------------+-----------+
| akismet             | inactive | available                    | 4.2.5     |
| hello               | inactive | none                         | 1.7.2     |
| some-extension-name | active   | none                         | 0.1.0     |
| woocommerce         | active   | version higher than expected | 6.8.0-dev |
| woocommerce-blocks  | active   | available                    | 8.2.0-dev |
+---------------------+----------+------------------------------+-----------+
```

opr committed:
Remove block registration code

opr commented:
Thanks @psealock for the review!

> I'm getting an email placed in the text box

This is fine, it's _under_ the box we don't want to see anything in the first step.

I've changed the template to remove the `register_newsletter_block_editor_blocks` function, seems this was causing it to double up.

Could you try again? By the way you might get two of the newsletter blocks, this is because of https://github.com/woocommerce/woocommerce-blocks/pull/6676 

Also, @ralucaStan this is blocked from being released by https://github.com/woocommerce/woocommerce-blocks/pull/6763 because we rely on the data store to show an example of validation handling in this block.

psealock commented:
I pulled down the latest Blocks `feature/data-store-refactor` branch and now I can't run `npm build` or `npm run dev`

<img width="939" alt="image" src="https://user-images.githubusercontent.com/1922453/183863545-0a334c78-5805-4be4-af5e-6ac0377dbbf3.png">

opr commented:
> I pulled down the latest Blocks `feature/data-store-refactor` branch and now I can't run `npm build` or `npm run dev`. I ran `npm install` beforehand.
> 
> <img alt="image" width="939" src="https://user-images.githubusercontent.com/1922453/183863545-0a334c78-5805-4be4-af5e-6ac0377dbbf3.png">

Weird! I just ran it and it worked OK for me. Did you get any more errors above this output, or in the debug log?

opr committed:
Register slug as a block category to prevent warnings in console

opr committed:
Add comment explaining the purpose of the function

psealock reviewed (dismissed):
This is testing well according to instructions, I'm seeing the newsletter signup

<img width="699" alt="image" src="https://user-images.githubusercontent.com/1922453/183879342-ba4ee8ee-401c-49b1-abef-f1ee42aff2c5.png">

opr commented:
Note: This shouldn't be merged until https://github.com/woocommerce/woocommerce-blocks/pull/6763 is merged

opr committed:
Bring templates up to date with latest changes to WC Blocks

The way the validation store selectors work was updated, so this template will use the new selectors.

opr committed:
Ensure error message shows when box is not checked

opr commented:
@psealock would you mind giving this another review please, it's had some updates since your last one. Thank you!

psealock reviewed (commented):
This is all working according to test instructions.

I do see some CSS a bit off on the edit side as there is a vertical center issue:

<img width="778" alt="image" src="https://user-images.githubusercontent.com/1922453/195735425-ced27bb0-d36c-4bc1-83c7-49ab6349fb48.png">

opr committed:
Ensure checkbox is targeted with CSS in editor

psealock commented:
@opr Just checking if you saw my comment above about CSS?

opr commented:
> @opr Just checking if you saw my comment above about CSS?
> 

Hey @psealock yep I've addressed that in 5b960e175d6b3aa45eb34f4b0366beb6deaf7fbe

Thanks for pointing it out!

psealock reviewed (approved):
With the latest CSS additions, this is testing as it should. I took a brief look at the code but don't have much context.
